react/features/device-selection/components/DeviceSelection.js

```diff
@@ -150,6 +162,25 @@ class DeviceSelection extends AbstractDialogTab<Props, State> {
         .then(() => this.props.mountCallback && this.props.mountCallback());
     }
 
+    /**
+     * Checks if audio / video permissions were granted.
+     *
+     * @param {Object} prevProps - Previous props this component received.
+     * @param {Object} prevState - Previous state this component had.
+     * @returns {void}
+     */
+    componentDidUpdate(prevProps, prevState) {
+        const { previewAudioTrack, previewVideoTrack } = prevState;
+
+        if ((!previewAudioTrack && this.state.previewAudioTrack)
+                || (!previewVideoTrack && this.state.previewVideoTrack)) {
+            this.props.hasAudioPermission()
+                .then(r => this.setState({ hasAudioPermission: r }));
+            this.props.hasVideoPermission()
+                .then(r => this.setState({ hasVideoPermission: r }));
```

Comment:
I also suppose two setState calls could potentially cause a double re-render or maybe more based on the timing of state getting set. Do you think the clarity of separate calls is better than a Promise.all or a modification to pass in one function instead of two or whatever other idea there might be?

Reply:
Good catch, I can fold this into a single update. Will fix.